go/sync2/semaphore.go

```diff
@@ -61,6 +62,19 @@ func (sem *Semaphore) Acquire() bool {
 	}
 }
 
+// AcquireContext returns true on successful acquisition, and
+// false on context expiry. Timeout is ignored.
+func (sem *Semaphore) AcquireContext(ctx context.Context) bool {
+	tm := time.NewTimer(sem.timeout)
```

Comment:
If timeout is ignored, why we need to create it? Seems like this could be removed.

Reply:
Good catch (copy-pasta). Fixed.